hoard/sources/whoas.py

```diff
@@ -163,16 +167,16 @@ def create_from_whoas_dim_xml(data: str, client: OAIClient) -> Dataset:
             kwargs["license"] = field.text
             kwargs["termsOfUse"] = field.text
 
-    kwargs["authors"] = authors
-    kwargs["contacts"] = contacts
-    kwargs["description"] = descriptions
     kwargs["subjects"] = ["Earth and Environmental Sciences"]
-    kwargs["distributors"] = distributors
-    kwargs["grantNumbers"] = grantNumbers
-    kwargs["keywords"] = keywords
-    kwargs["language"] = languages
-    kwargs["notesText"] = notesText
-    kwargs["otherIds"] = otherIds
-    kwargs["publications"] = publications
-    kwargs["timePeriodsCovered"] = timePeriodsCovered
+    if notesText != "":
+        kwargs["notesText"] = notesText
+    if "title" not in kwargs:
+        logger.info(f"dc.title not found in record: {rec_id}")
+        kwargs["title"] = None
+    elif "authors" not in kwargs:
+        logger.info(f"dc.contributor.author not found in record: {rec_id}")
+        kwargs["authors"] = None
+    elif "description" not in kwargs:
+        logger.info(f"dc.description.abstract not found in record: {rec_id}")
+        kwargs["description"] = None
```

Comment:
As per the comment above, none of these lines should be necessary. The creation of a dataset should throw an exception if these values don't exist. The place to log anything would be in the WHOAS iterator when the exception is thrown. If we are indeed just going to skip invalid records, then catch the exception, log the exception message and continue iterating.